src/parse/state/tag.ts

```diff
@@ -383,6 +383,11 @@ function read_attribute(parser: Parser, unique_names: Set<string>) {
 	if (parser.eat('=')) {
 		value = read_attribute_value(parser);
 		end = parser.index;
+	} else if (parser.match_regex(/["']/)) {
+		parser.error({
+			code: `expected-equals`,
```

Comment:
I'm leery of introducing a new error code just for this. Is there some more generic 'unexpected character' error that could be used instead?

Reply:
Alright. We could also use the generic `unexpected-token` from `eat`:

```javascript
this.error({
  code: `unexpected-${this.index === this.template.length ? 'eof' : 'token'}`,
  message: message || `Expected ${str}`
});
```

If the new error handling code is removed, the following error is thrown:

```html
5: <h1 class"=foo">Hello {name}!</h1>
            ^
ParseError: Expected >
```
That might be good enough?